lib/auth/auth.go

```diff
@@ -3205,6 +3205,7 @@ func generateCert(ctx context.Context, a *Server, req certRequest, caType types.
 			Roles:                   req.checker.RoleNames(),
 			CertificateFormat:       certificateFormat,
 			PermitPortForwarding:    req.checker.CanPortForward(),
+			SSHPortForwardMode:      req.checker.SSHPortForwardMode(),
```

Comment:
let's say that my user has a single role with
```
ssh_port_forwarding:
  remote: false
```

CanPortForward should return false here and the PermitPortForwarding extension in the cert should be unset. That's all good. I'm wondering if an older ssh node that doesn't know about the new role fields will respect that the extension is unset and block port forwarding, or if it ignores the cert and it will just allow port forwarding. Have you tested this out, or can you, and let us know here? Ideally I think we should try to prevent port forwarding on older nodes in that case

Reply:
As far as I can tell we don't reference the cert extension anywhere other than to set it. I'm doing another round of manual testing though so I'll double check that this works as expected :+1:

Reply:
let us know how the testing goes, you're right I don't think we reference it anywhere, but there's a chance that `golang.org/x/crypto/ssh` will honor it. But I think if you can port-forward to older nodes with my original example role we should try to find a mitigation (like downgrading the role for those nodes)

Reply:
I tested with a v16 agent and it allowed me to port forward even with `ssh_port_forwarding` set to false across the board, which is what we expected to happen. I've got another PR ready to go behind this one that currently downgrades the role for older agents.

I actually think we can get away with _always_ setting `port_forwarding` appropriately rather than doing an optional downgrade based on version. Newer agents will ignore values for `port_forwarding` in favor of `ssh_port_forwarding` which should mean`port_forwarding` is free to be leveraged as a way to signal correct behavior to older agents.